src/userobjects/OpenMCUserObject.C

```diff
@@ -0,0 +1,57 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCUserObject.h"
+#include "OpenMCProblemBase.h"
+
+InputParameters
+OpenMCUserObject::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  return params;
+}
+
+OpenMCUserObject::OpenMCUserObject(const InputParameters & parameters)
+  : GeneralUserObject(parameters), _first_execution(true)
+{
+  if (!openmc_problem())
+  {
+    std::string extra_help = _fe_problem.type() == "FEProblem" ? " (the default)" : "";
+    mooseError("This user object can only be used with wrapped OpenMC cases! "
+               "You need to change the\nproblem type from '" +
+               _fe_problem.type() + "'" + extra_help + " to OpenMCCellAverageProblem.");
+  }
+}
+
+void
+OpenMCUserObject::execute()
```

Comment:
Why is this method defined here? As far as I can tell, the only two derived classes of this base class are `OpenMCDomainFilterEditor` and `OpenMCTallyEditor`, both of which override `execute()` and do not call the parent class method at all. It seems like this `OpenMCUserObject::execute()` method never gets called.

Is this the case, or am I missing something?

Reply:
You...are not missing something. This feels like a holdover from when I allowed the tally editors to be attached to the Cardinal tallies. I'm going to look into the control flow of this again.